dashboard/src/actions/datasetListActions.js

```diff
@@ -1,27 +1,76 @@
 import * as TYPES from "./types";
 
 import API from "../utils/axiosInstance";
+import { DANGER } from "assets/constants/toastConstants";
+import { showToast } from "./toastActions";
 import { uriTemplate } from "utils/helper";
 
-export const fetchPublicDatasets = () => async (dispatch, getState) => {
+export const fetchPublicDatasets = (page) => async (dispatch, getState) => {
   try {
     dispatch({ type: TYPES.LOADING });
     const endpoints = getState().apiEndpoint.endpoints;
+    const { offset, limit, filter, searchKey, perPage } =
+      getState().datasetlist;
+    let publicData = [...getState().datasetlist.publicData];
+    const params = new URLSearchParams();
+    params.append("metadata", "dataset.uploaded");
+    params.append("access", "public");
+    params.append("offset", offset);
+    params.append("limit", limit);
+
+    if (searchKey) {
+      params.append("name", searchKey);
+    }
+    if (filter.startDate instanceof Date && !isNaN(filter.startDate)) {
+      params.append("start", new Date(filter.startDate).toUTCString());
+    }
+    if (filter.endDate instanceof Date && !isNaN(filter.endDate)) {
+      params.append("end", new Date(filter.endDate).toUTCString());
+    }
+
     const response = await API.get(
       uriTemplate(endpoints, "datasets_list", {}),
-      { params: { metadata: "dataset.uploaded", access: "public" } }
+      {
+        params,
+      }
     );
+
     if (response.status === 200 && response.data) {
+      const startIdx = (page - 1) * perPage;
+
+      if (publicData.length === 0) {
+        publicData = new Array(response.data.total);
+      }
```

Comment:
Another path to consider: in Javascript, the `length` property of an array is *writeable*, so `publicData.length = response.data.total` will set the length of the array to the total number of datasets, whether that's an increase or decrease.

Weird. 😆

Reply:
Oh, I like that so much better than my suggestion!

Reply:
> Oh, I like that so much better than my suggestion!

I should be specific:  I like that suggestion for the cases where the length of the array has not shrunk.  It doesn't help if the array has shrunk.

Unfortunately, now that I think about it, it doesn't help in cases where the array has grown, either, because we don't know where in the array the new entry was added (and, because of that, I think we can end up with gaps and overlaps due to the fact that old pages previously loaded are no longer aligned with newly-loaded pages...basically the same problem which results from the array shrinking).

So, I think there are only two cases:
- the length of the array from the Redux state matches the `total`, in which case we can use it as is, or
- the length is "wrong" (empty or not), in which case we need to create a new (sparse) array of the appropriate size.

Reply:
Updated the size of the array as per suggestions